specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-11-05-preview/Migrations.json

```diff
@@ -0,0 +1,655 @@
+        "sourceDBServerResourceId": {
+          "type": "string"
+        },
+        "secretParameters": {
+          "$ref": "#/definitions/MigrationSecretParameters"
+        },
+        "targetDBServerResourceId": {
+          "type": "string"
+        },
+        "targetDBServerSubnetResourceId": {
+          "type": "string"
+        },
+        "dBsToMigrate": {
+          "maxItems": 8,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "migrationResourceGroup": {
+          "$ref": "#/definitions/MigrationResourceGroup"
+        },
+        "setupLogicalReplicationOnSourceDBIfNeeded": {
+          "type": "boolean"
+        },
+        "overwriteDBsInTarget": {
+          "type": "boolean"
+        },
+        "migrationWindowStartTimeInUtc": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "startDataMigration": {
+          "type": "boolean"
+        },
+        "triggerCutover": {
+          "type": "boolean"
+        },
+        "cancel": {
+          "type": "boolean"
+        }
+      },
+      "description": "Migration resource properties."
+    },
+    "MigrationResourceListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MigrationResource"
+          },
+          "description": "The list of migration resources"
+        }
+      },
+      "description": "A list of migration resources."
+    },
+    "MigrationResourceForPatch": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MigrationResourcePropertiesForPatch",
+          "description": "Migration resource properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "Represents a migration resource for patch."
+    },
+    "MigrationResourcePropertiesForPatch": {
+      "type": "object",
+      "properties": {
+        "setupLogicalReplicationOnSourceDBIfNeeded": {
+          "type": "boolean"
+        },
+        "overwriteDBsInTarget": {
+          "type": "boolean"
+        },
+        "migrationWindowStartTimeInUtc": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "startDataMigration": {
+          "type": "boolean"
+        },
+        "triggerCutover": {
+          "type": "boolean"
+        },
+        "cancel": {
+          "type": "boolean"
+        }
+      },
+      "description": "Migration resource properties for patch."
+    },
+    "MigrationResourceGroup": {
+      "required": [
+        "resourceId"
+      ],
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "type": "string"
+        },
+        "location": {
+          "maxLength": 128,
+          "type": "string"
+        }
+      },
+      "description": "Migration resource group."
+    },
+    "MigrationSecretParameters": {
```

Comment:
what are these used for and why can't Managed Resource Identity be used instead?

Reply:
This is a private preview version. Will move to Managed identity in public preview.